test/AutoDiff/differential_operators.swift

```diff
@@ -6,20 +6,23 @@ import StdlibUnittest
 
 var BuiltinDifferentialOperatorTests = TestSuite("BuiltinDifferentialOperators")
 
-@inlinable
-func valueWithPullback<T, R>(
-  at x: T, in f: @autodiff (T) -> R
-) -> (R, (R.CotangentVector) -> T.CotangentVector)
-  where T : Differentiable, R : Differentiable {
-  return Builtin.autodiffApplyVJP(f, x)
-}
-
 BuiltinDifferentialOperatorTests.test("Trivial") {
   let t = 1.0
-  let (y, pullback) = valueWithPullback(at: 4.0) { x in
+  let (value: y, pullback: pb) = valueWithPullback(at: 4.0) { x in
+    x * x * t
+  }
+  expectEqual(16, y)
+  expectEqual(8, pb(1))
+
+  let pb2 = pullback(at: 4.0) { x in
+    x * x * t
+  }
+  expectEqual(8, pb2(1))
+
+  let grad = gradient(at: 4.0) { x in
     x * x * t
   }
-  expectEqual(8, pullback(1))
+  expectEqual(8, grad)
```

Comment:
could you also add a test for `valueWithGradient`?

Reply:
Added.